src/gc.c

```diff
@@ -2805,19 +2807,46 @@ JL_DLLEXPORT void *jl_realloc(void *p, size_t sz)
     return (void *)(pnew + 2);
 }
 
-JL_DLLEXPORT void *jl_gc_managed_malloc(size_t sz)
+// TODO add special casing for macOS, where there is guarantee of alignment
+// TODO add special casing for 64 bit systems when 16 byte alignment is requested
+// TODO add checks on align?
+void *jl_gc_malloc_aligned(size_t sz, size_t align)
 {
-    jl_ptls_t ptls = jl_get_ptls_states();
-    maybe_collect(ptls);
-    size_t allocsz = LLT_ALIGN(sz, JL_CACHE_BYTE_ALIGNMENT);
-    if (allocsz < sz)  // overflow in adding offs, size was "negative"
-        jl_throw(jl_memory_exception);
-    gc_num.allocd += allocsz;
-    gc_num.malloc++;
-    void *b = malloc_cache_align(allocsz);
-    if (b == NULL)
-        jl_throw(jl_memory_exception);
-    return b;
+    void *p0 = jl_gc_malloc(sz + align);
```

Comment:
Right, I'm pretty sure I left that comment somewhere but couldn't find it. It was incorrectly introduced in https://github.com/JuliaLang/julia/commit/3e2d34362acd4d00e9b3ed8e2dda8601415fdddd#diff-31b93f4272eaeede3738531e90317e0cL2674 More importantly, the `calloc` version over-allocate a lot.